Snowflake.Data/Core/QueryContextCache.cs

```diff
@@ -221,26 +228,29 @@ public int GetSize()
          */
         public void Update(ResponseQueryContext queryContext)
         {
-            // Log existing cache entries
-            LogCacheEntries();
-
-            if (queryContext == null || queryContext.Entries == null)
-            {
-                // Clear the cache
-                ClearCache();
-                return;
-            }
-            foreach (ResponseQueryContextElement entry in queryContext.Entries)
+            lock(_qccLock)
             {
-                Merge(entry.Id, entry.ReadTimestamp, entry.Priority, entry.Context);
-            }
+                // Log existing cache entries
+                LogCacheEntries();
```

Comment:
Does it make sense to replace #ifdef with check for _logger.IsDebugEnabled()?
We could see this in the logs even using a production code.

Reply:
fixed.

Reply:
@sfc-gh-mhofman  sorry for the delay please review.